### PR TITLE
:bug: PIL Image.open was never closed in workspace, #377

### PR DESCRIPTION
**Pull request by kba**

### CODE REVIEW

File: ocrd/ocrd/workspace.py [hunk 1]
```diff
@@ -215,7 +215,9 @@ def resolve_image_exif(self, image_url):
 
         if image_url not in self.image_cache['exif']:
             # FIXME must be in the right directory
-            self.image_cache['exif'][image_url] = OcrdExif(Image.open(image_filename))
+            with Image.open(image_filename) as pil_img:
```
Comment (wrznr) [inline ocrd/ocrd/workspace.py]: While the changes might be useful, they do not solve the initial problem which was reported for line https://github.com/OCR-D/core/pull/390/files#diff-c9ac3688a5769ae00495937cd8b76233L185

```
Traceback (most recent call last):
  File "/home/kmw/Documents/Work/OCR-D/env/bin/ocrd-cis-ocropy-denoise", line 8, in <module>
  File "/home/kmw/Documents/Work/OCR-D/env/lib/python3.6/site-packages/click/core.py", line 764, in __call__
  File "/home/kmw/Documents/Work/OCR-D/env/lib/python3.6/site-packages/click/core.py", line 717, in main
  File "/home/kmw/Documents/Work/OCR-D/env/lib/python3.6/site-packages/click/core.py", line 956, in invoke
  File "/home/kmw/Documents/Work/OCR-D/env/lib/python3.6/site-packages/click/core.py", line 555, in invoke
  File "/home/kmw/Documents/Work/OCR-D/env/lib/python3.6/site-packages/ocrd_cis/ocropy/cli.py", line 29, in ocrd_cis_ocropy_denoise
  File "/home/kmw/Documents/Work/OCR-D/env/lib/python3.6/site-packages/ocrd/decorators.py", line 54, in ocrd_cli_wrap_processor
  File "/home/kmw/Documents/Work/OCR-D/env/lib/python3.6/site-packages/ocrd/processor/base.py", line 56, in run_processor
  File "/home/kmw/Documents/Work/OCR-D/env/lib/python3.6/site-packages/ocrd_cis/ocropy/denoise.py", line 138, in process
  File "/home/kmw/Documents/Work/OCR-D/env/lib/python3.6/site-packages/ocrd/workspace.py", line 185, in add_file
OSError: [Errno 24] Too many open files: 'DENOISE3/DENOISE3_0668.xml'
```

Comment (kba) [reply ocrd/ocrd/workspace.py]: > While the changes might be useful, they do not solve the initial problem which was reported for line

That line isn't the problem, it's a standard `with open(...)`, this is just the one file opened too many.

Comment (wrznr) [reply ocrd/ocrd/workspace.py]: Granted. But still, the problem persists.

**Comment by kba** [comment]
I'm still investigating the original issue but there is no reason not to close this error source now.

Tracking progress on the too-many-files-open issue in #377 as I'm pretty sure that's the root of the issue.

**Pull request merged**